pkg/osv/osv.go

```diff
@@ -158,7 +158,7 @@ func chunkBy[T any](items []T, chunkSize int) [][]T {
 
 // checkResponseError checks if the response has an error.
 func checkResponseError(resp *http.Response) error {
-	if resp.StatusCode == http.StatusOK {
+	if resp.StatusCode >= 200 && resp.StatusCode < 300 {
```

Comment:
suggestion: I think we shouldn't change this line, since it's not needed for addressing #861

even though 2xx are success codes, we're only expecting the API currently to return a 200 and since this change isn't strictly needed for the rest of the change to work, I would lean towards not changing it for now

Reply:
I've been cargo-culting the HTTP retry logic from [the examples in https://github.com/sethvargo/go-retry/blob/main/retry_test.go](https://github.com/sethvargo/go-retry/blob/main/retry_test.go#L175-L228), for what it's worth, which treats 5xx errors as retryable, only. That said, I also recently saw https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429 in the wild, which could be retried with an exponential backoff approach.